src/array/growable/mod.rs

```diff
@@ -40,6 +40,13 @@ pub trait Growable<'a> {
     /// Extends this [`Growable`] with null elements, disregarding the bound arrays
     fn extend_validity(&mut self, additional: usize);
 
+    /// Extends this [`Growable`] by iterator.
+    fn extend_from_iter(&mut self, iter: Box<dyn Iterator<Item = (usize, usize, usize)>>) {
```

Comment:
Yeap, I think that this won't work as intended because `.extend` will still be dynamically dispatched.

We do something like what you intend here: https://github.com/jorgecarleitao/arrow2/blob/main/src/compute/filter.rs#L87

Essentially, specialize the implementation based on the physical type, and use a generic one (https://github.com/jorgecarleitao/arrow2/blob/main/src/compute/filter.rs#L101) for all others.

Reply:
https://users.rust-lang.org/t/is-this-a-curiously-recurring-template-pattern/31004/6

As User @ExpHP said:

```
trait Trait {
    fn foo(&mut self);
    fn call_foo_twice(&mut self) {
        self.foo(); // <-- this is statically dispatched
        self.foo();
    }

    fn wrap(self) -> Wrapper<Self> { // <-- we can name Self
        Wrapper(self)
    }
}
```

so `.extend ` is statically dispatched?